.github/workflows/ci_gcc4.yml

```diff
@@ -0,0 +1,34 @@
+name: GCC 4 CI
```

Comment:
almost identical to `.github/workflows/ci_packages.yml`, how about refactor what's common and generate both via tools/ci_generate.nim, as done with other CI's

Reply:
For Github we can template our CI natively via [composite run steps](https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action) instead of generating these manually. The main perk would be that we can use those templated steps outside of this repo as well.